code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -4,6 +4,25 @@
 	flags = 0 //starts closed
 	drop_sound = 'sound/items/drop/soda.ogg'
 	pickup_sound = 'sound/items/pickup/soda.ogg'
+	desc_info = "Click it in your hand to open it.\
+				 If it's carbonated and closed, you can shake it by clicking on it with harm intent. \
+				 If it's empty, you can crush it on your forehead by selecting your head and clicking on yourself with harm intent. \
+				 You can also crush cans on other people's foreheads as well."
+
+/obj/item/reagent_containers/food/drinks/cans/attack(mob/living/M, mob/user, var/target_zone)
+	if(istype(M, /mob/living/carbon) && !reagents.total_volume && user.a_intent == I_HURT && target_zone == BP_HEAD)
+		if(M == user)
+			user.visible_message(SPAN_WARNING("[user] crushes the can of [src.name] on [user.get_pronoun(1)] forehead!"), SPAN_NOTICE("You crush the can of [src.name] on your forehead."))
+		else
```

Comment:
I can already imagine the misclicks, please just make this crush the can in their hands instead LOL

Reply:
what do you mean by misclick?

by misclick as in misinterpreting the intstructions
anyway, i added a desc_info that i felt was pretty clear, so people can execute it pretty decently.

by misclick as in, mistake, 
in any case, you'd spill the drink on yourself first anyway.
well, intentional? you need it to be empty to crush it on your head anyway.